packages/@aws-cdk/aws-dynamodb/lib/table-time-to-live-handler/utils.ts

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable no-console */
+import type { OnEventRequest } from '@aws-cdk/custom-resources/lib/provider-framework/types';
+import { AWSError, DynamoDB } from 'aws-sdk'; // eslint-disable-line import/no-extraneous-dependencies
+const dynamodb = new DynamoDB({ apiVersion: '2012-08-10' });
+
+
+export async function disableTimeToLive(event: OnEventRequest) {
+  const ttl = await dynamodb.describeTimeToLive({ TableName: event.ResourceProperties.TableName }).promise();
+  await dynamodb.updateTimeToLive({
+    TableName: event.ResourceProperties.TableName,
+    TimeToLiveSpecification: {
+      AttributeName: ttl.TimeToLiveDescription?.AttributeName ?? '',
+      Enabled: false,
+    },
+  }).promise();
+}
+
+export async function enableTimeToLive(event: OnEventRequest) {
+  try {
+    await dynamodb.updateTimeToLive({
+      TableName: event.ResourceProperties.TableName,
+      TimeToLiveSpecification: {
+        AttributeName: event.ResourceProperties.TimeToLiveAttribute,
+        Enabled: true,
+      },
+    }).promise();
+  } catch (err) {
+    // Catch exception so we can try enabling again at a later point in time.
+    // This is necessary if the ttl was just disabled as enabling is not available right away.
+    const awsError = err as AWSError;
+    if (
+      awsError.code === 'ValidationException' &&
+      awsError.message === 'Time to live has been modified multiple times within a fixed interval'
+    ) {
+      console.log('Time to live has been modified multiple times within a fixed interval. Try again later.');
+      return;
```

Comment:
Doesn't catching the error here means that the custom resource will succeed even though the TTL was not correctly updated?

Reply:
No as the correctness check in index will still return false. The worst-case scenario is the custom resource will timeout. We need to do this to allow changing from MyTTL to MyNewTTL. If we disable it, the api will return that the ttl is disabled, but won’t let us set a new ttl until all things are propagated in the background. As we don’t (or at least I don’t) know how long this will take, the only option is to retry in intervals.

Reply:
I understand. Maybe it's better to make the custom resource always fail if `updateTimeToLive()` throws? Do you happen to know how long is this `fixed interval`?

Let's wait for @RomainMuller's feedback on this one...

Reply:
This absolutely deserves a comment in the code :).

Also, note that you don't catch the exception in `disableTimeToLive`. So, editing when this interval is not available still fails (as the first step of editing a TTL is disabling it).

I would vote for not catching this exception, and just failing, like we do in `disableTimeToLive`. This way, the customer will have to know to wait with the update until DynamoDB allows it again.

Either that, or catch this also in `disableTimeToLive`.

Reply:
@skinny85 Please let me know if the comment is clear enough. For ease of use I would like to catch this block as otherwise the enabled -> disabled -> enabled flow (needed for updating without manually disabling) won’t work. To mimic the behavior I have added a catch to `disableTimeToLive(...)`.

Please see also https://github.com/aws/aws-cdk/pull/13720#issuecomment-813317752